tbot_contrib/locking.py

```diff
@@ -0,0 +1,166 @@
+import contextlib
+import typing
+import tbot
+import time
+from tbot.machine import machine, linux
+
+
+class MachineLock(machine.PreConnectInitializer):
+    """
+    Machine locking implementation based on Python and bash.
+    """
+    @property
+    def lock_name(self) -> str:
+        return self.__class__.__name__
+
+    @property
+    def lock_expiry(self) -> int:
+        return -1
+
+    @property
+    def lock_pool_max(self) -> int:
+        return 1
+
+    @property
+    def lock_sleep(self) -> int:
+        return 1
+
+    lock_no = -1
+    lock_file = ""
```

Comment:
If I understand correctly, those are actually instance attributes, not class attributes.  As such, please don't define them on the class.